src/Grid/DataGrid/DataGrid.jsx

```diff
@@ -1,193 +1,112 @@
-import Table from 'antd/lib/table';
-import Form from 'antd/lib/form';
-import Popconfirm from 'antd/lib/popconfirm';
-import Icon from 'antd/lib/icon';
-import Tooltip from 'antd/lib/tooltip';
-import { Column } from './Column';
-import Button from 'antd/lib/button';
-import { message } from 'antd';
-import nanoid from 'nanoid';
-
-export const EditableContext = React.createContext();
-
-const ClickableIcon = styled(Icon)`
-    cursor: pointer;
-    font-size: 20px;
-    margin: 0 10px;
-`;
+import { message, Form } from 'antd';
+import { BaseGrid, EditableContext } from '../BaseGrid';
+import { IconButton, emptyFn } from '../..';
 
 /**
  * @return {React.Component}
  *
  * @constructor
  */
 export const DataGrid = Form.create()((props) => {
-    const {idProperty, dataSource, onSave, onDelete, onRecordCreate, form, children, ...restProps} = props;
-    const [editingKey, setEditingKey] = useState('');
-    const [data, setData] = useState(dataSource);
-
-    useEffect(() => {
-        setData(dataSource);
-    }, [dataSource]);
-
-    const isEditing = record => record[idProperty] === editingKey;
-
-    const onCancelClick = () => {
-        setEditingKey('');
-    };
+    const { idProperty, onAdd, onEdit, onSave, form, children, ...restProps } = props;
+    const [isEditing, setEditing] = useState(false);
+    const [selectedRowKeys, setSelectedRowKeys] = useState([]);
+    const [isLoading, setLoading] = useState(false);
 
-    const onEditClick = (key) => {
-        setEditingKey(key);
-    };
-
-    const onSaveClick = (form, key) => {
-        form.validateFields((error, row) => {
+    const onSaveClick = () => {
+        form.validateFields((error, record) => {
             if (error) {
-                return;
+                return message.error('form validation failed');
             }
-            const newData = [...data];
-            const index = newData.findIndex(item => key === item[idProperty]);
-            if (index > -1) {
-                const item = newData[index];
-                newData.splice(index, 1, {
-                    ...item,
-                    ...row,
+
+            setLoading(true);
+
+            onSave({ [idProperty]: selectedRowKeys[0], ...record })
+                .then(() => {
+                    setLoading(false);
+                    setEditing(false);
+                })
+                .catch(err => {
+                    setLoading(false);
+                    message.error(err.toString());
                 });
-                saveData(newData);
-                setEditingKey('');
-            } else {
-                newData.push(row);
-                saveData(newData);
-                setEditingKey('');
-            }
         });
     };
 
-    const onAddRowClick = () => {
-        const record = onRecordCreate({[idProperty]: nanoid(10)});
-        const newData = [...data, record];
-        saveData(newData);
+    const onAddClick = (rec) => {
```

Comment:
das sieht irgendwie seltsam aus. was passiert hier?

Reply:
was meinst? du kannst übrigens auch mehrere zeilen markieren und diese dann kommentieren

Reply:
du speicherst das onAdd(rec) in record. und dann im greifst du über die idProperty rauf zu.. was gibt denn onAdd zurück?

Reply:
du kannst in der `onAdd` methode ein object zurückgeben um z.b. einige felder im form initial zu befüllen. da war aber ein fehler drin, denn das zugreifen auf die idProperty gehört da nicht hin sondern ins BaseGrid (war da auch schon drin)